lib/std/special/compiler_rt/README.md

```diff
@@ -1,15 +1,257 @@
-This compiler-rt library is ported from [LLVM](http://compiler-rt.llvm.org/).
+If hardware lacks basic or specialized functionality, compiler-rt adds such functionality
+for basic arithmetic(s).
+One such example is 64-bit integer multiplication on 32-bit x86.
 
-It's needed because LLVM emits library calls to compiler-rt when hardware lacks
-functionality, for example, 64-bit integer multiplication on 32-bit x86.
+Current status (tracking libgcc documentation):
+- Integer library routines => almost implemented
+- Soft float library routines => only f80 routines missing
+- Decimal float library routines => unimplemented (~120 functions)
+- Fixed-point fractional library routines => unimplemented (~300 functions)
+- Exception handling routines => unclear, if supported (~32+x undocumented functions)
+- Miscellaneous routines => unclear, if supporetd (cache control and stack function)
 
 This library is automatically built as-needed for the compilation target and
 then statically linked and therefore is a transparent dependency for the
 programmer.
+For details see `../compiler_rt.zig`.
 
-Any bugs should be solved by trying to duplicate the bug upstream.
- * If the bug exists upstream, get it fixed with the LLVM team and then port
-   the fix downstream to Zig.
- * If the bug only exists in Zig, something went wrong porting the code,
-   and you can run the C code and Zig code side by side in a debugger
-   to figure out what's happening differently.
+The routines in this folder are listed below.
+Routines are annotated as `type source routine // desription`, with `routine`
+being the name used in aforementioned `compiler_rt.zig`.
+`dev` means deviating from compiler_rt, `port` ported, `source` is the
+information source for the implementation, `none` means unimplemented.
+In the following a brief overview of is given:
```

Comment:
I'm having difficult parsing this sentence - I think you may be missing a word after "of" ("In the following" also seems perhaps redundant)?

Reply:
good catch.